subsys/net/lib/dhcpv4/dhcpv4_server.c

```diff
@@ -1231,7 +1231,7 @@ int net_dhcpv4_server_start(struct net_if *iface, struct in_addr *base_addr)
 		return -EINVAL;
 	}
 
-	netmask = net_if_ipv4_get_netmask(iface);
+	netmask = net_if_ipv4_get_netmask_by_addr(iface, base_addr);
```

Comment:
I think `server_addr` should be used here instead. `base_addr` won't be configured as IPv4 address on the interface.

Reply:
Indeed, will fix.